public/ServerConfig.json

```diff
@@ -1,3 +1,15 @@
 {
-    "baseURL": "http://localhost:3001/api"
+    "baseURL": "http://localhost:3001/api",
+    "fhir" : {
+        "shimServerOverrides": {
```

Comment:
The javascript smart client is a global object and does not act like an instance but rather a singleton.    There may be a way to remove the use of the config file and move the information into the meta data contained in the app manager but it will require some reworking.  

This can be overridden in parameters passed into the window.FHIR.oauth2.authorize method being called in the flux Launch component .  The shim server already supports returning the oauth2 urls required for authenticating to the server so we would need to redirect the client to use the shim server as a baseurl in the Launch component.    Mapping the iss params passed in to a specific shim server depending on the Flux app(s) configured to use Smart on FHIR will require either  multiple Launch endpoints with the override provided in the metadata in AppManager or a single Launch endpoint that uses a config file such as ServerConfig that contains the mappings.   This may require similar mappings for the flux SmartApp as well.  

I have no real preference on either way just as long as it is testable.  We can sync up and decide on which is the best approach to take.

Reply:
Sorry. I'm not quite sure I followed everything in your comments so a discussion probably makes sense. @dehall can you set up a time for the 3 of us to talk? Actually, let's make it 4 and include Dan. Definitely add a skype if it's today! :) Basically, I think the shim should be a SMART on FHIR server and that Flux Notes should not be aware of whether it's talking to the shim or some EHR that supports mCODE.